domain/schema/model/sql/0014-constraint.sql

```diff
@@ -14,9 +14,20 @@ CREATE TABLE "constraint" (
     image_id TEXT,
     CONSTRAINT fk_constraint_container_type
     FOREIGN KEY (container_type_id)
-    REFERENCES container_type (id)
+    REFERENCES constraint_container_type (id)
 );
 
+-- constraint_container_type represents the valid container types that a
+-- constraint can have.
+CREATE TABLE constraint_container_type (
```

Comment:
I wonder if we want just "container_type"?
Yes, the constraint attribute uses it, but the underlying type is used for provisioning etc. If we want to know the container type of a machine, we could add it as a FK there. The argument that we could get there indirectly via looking at a machine's constraint assumes we record that against the machine which we don't. And there may well be other artefacts that link to container type.

Reply:
yeah I am happy for that. I just wasn't sure where this should go and if I am modelling another part of the system that hasn't been designed yet. Where do we want to put this ?

Reply:
I'd add it to machine.sql for now

Reply:
righto, ta